src/GraphQL.NewtonsoftJson/GraphQLRequestListJsonConverter.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.Collections.Generic;
+using GraphQL.Transport;
+using Newtonsoft.Json;
+
+namespace GraphQL.NewtonsoftJson
+{
+    /// <summary>
+    /// A custom JsonConverter for reading or writing a <see cref="GraphQLRequest"/> object.
+    /// </summary>
+    public class GraphQLRequestListJsonConverter : JsonConverter
+    {
+        /// <inheritdoc/>
+        public override bool CanConvert(Type objectType)
+        {
+            return (
+                objectType == typeof(IEnumerable<GraphQLRequest>) ||
+                objectType == typeof(ICollection<GraphQLRequest>) ||
+                objectType == typeof(IReadOnlyCollection<GraphQLRequest>) ||
+                objectType == typeof(IReadOnlyList<GraphQLRequest>) ||
+                objectType == typeof(IList<GraphQLRequest>) ||
+                objectType == typeof(List<GraphQLRequest>) ||
+                objectType == typeof(GraphQLRequest[]));
```

Comment:
move this one into the top for performance

Reply:
The least allocations would be to use the `IList` variation with a `for` loop.  This is because for a single request, an array of one element can be allocated during deserialization and passed as an `IList` to the server, where the length can be read and the single element pulled out.  For batch requests, a `List` is created during deserialization.  If it were to convert to an array, it would need to allocate another array and copy the elements to it.  Rather as an `IList` it can return the `List` directly and the server project can read the length and iterate over the items with `for`.

I'll move the `IList` to the top, followed by `[]`, `IEnumerable` and `List`, then the others.

Reply:
Done as indicated

Reply:
ok